### PR TITLE
Aerie 2.18.0

### LINKED ISSUE #146: Assert Aerie version
Tied to NASA-AMMOS/aerie#1535, check against an Aerie endpoint and hard-code version compatibility. Report an error or warning to the user if the Aerie host is on an incompatible version. Don't report an error if the host is on `develop`, or maybe report only a warning.

### PR DESCRIPTION
**Pull request by cartermak**
Update to Aerie 2.18.0 and implement a check that the Aerie host has a compatible version. 

Closes #146

### CODE REVIEW

File: tests/unit_tests/test_aerie_host.py
Comment (cartermak) [inline tests/unit_tests/test_aerie_host.py:1]: This file is a bit of a mess -- it'd be good to come back & clean up with some more consistent mocks.

File: src/aerie_cli/aerie_host.py [hunk 1]
```diff
@@ -260,7 +264,15 @@ def is_auth_enabled(self) -> bool:
...
-    def authenticate(self, username: str, password: str = None):
+    def authenticate(self, username: str, password: str = None, override: bool = False):
+
+        try:
+            self.check_aerie_version()
+        except AerieHostVersionError as e:
+            if override:
+                print("Warning: " + e.args[0])
+            else:
+                raise
```
Comment (Mythicaeda) [inline src/aerie_cli/aerie_host.py:275]: Shouldn't this say `raise e`?

File: src/aerie_cli/aerie_host.py [hunk 2]
```diff
@@ -18,15 +25,12 @@ def process_gateway_response(resp: requests.Response) -> dict:
...
-        raise RuntimeError(f"Bad response from Aerie Gateway.")
+        raise RuntimeError("Bad response from Aerie Gateway")
 
     try:
         resp_json = resp.json()
     except requests.exceptions.JSONDecodeError:
-        raise RuntimeError(f"Failed to get response JSON")
-
-    if "success" in resp_json.keys() and not resp_json["success"]:
-        raise RuntimeError(f"Aerie Gateway request was not successful")
```
Comment (Mythicaeda) [inline src/aerie_cli/aerie_host.py:29]: You ought to move this check out to when you're hitting the `login` endpoint.

File: src/aerie_cli/aerie_host.py [hunk 3]
```diff
@@ -278,6 +290,28 @@ def authenticate(self, username: str, password: str = None):
...
+
+        resp = self.session.get(self.gateway_url + "/version")
+
+        try:
+            resp_json = process_gateway_response(resp)
+            host_version = resp_json["version"]
+        except (RuntimeError, KeyError):
+            # If the Gateway responded, the route doesn't exist
+            if resp.text and "Aerie Gateway" in resp.text:
+                raise AerieHostVersionError("Incompatible Aerie version: host version unknown")
```
Comment (Mythicaeda) [inline src/aerie_cli/aerie_host.py:307]: Could you share an example response for when you try to hit this endpoint on an Aerie version without a `version` endpoint?

File: src/aerie_cli/utils/sessions.py [hunk 4]
```diff
@@ -136,6 +137,7 @@ def start_session_from_configuration(
         username (str, optional): Aerie username.
         password (str, optional): Aerie password.
         secret_post_vars (Dict[str, str], optional): Optionally provide values for some or all secret post request variable values. Defaults to None.
+        override (bool, optional): Override Aerie host version check. Defaults to False.
```
Comment (Mythicaeda) [inline src/aerie_cli/utils/sessions.py:140]: Nonblocking, feel free to resolve: the name feels a little confusing, as I would expect an override to be setting a value, not ignoring the results of a check. Maybe `force`, as that feels similar to how for other bash CLIs, the `force/f` flag is used to forcibly execute a command that would normally except?

File: tests/unit_tests/test_aerie_host.py
Comment (Mythicaeda) [reply tests/unit_tests/test_aerie_host.py:1]: Do you intend to do this as part of this PR?

File: src/aerie_cli/aerie_host.py (hunk 3 above)
Comment (cartermak) [reply src/aerie_cli/aerie_host.py:307]: image

A `get` request to an unknown route on the Gateway just gets this regular UI page back (interesting that it doesn't get a 404 or even a redirect, but 🤷)

Comment (Mythicaeda) [reply src/aerie_cli/aerie_host.py:307]: Huh. Alright I see why you're checking for `Aerie Gateway` in the response text now

File: src/aerie_cli/aerie_host.py (hunk 1 above)
Comment (cartermak) [reply src/aerie_cli/aerie_host.py:275]: Having `raise` in an `except` just re-raises the last exception that was caught, so it should be equivalent. The unit tests exercise this

Comment (Mythicaeda) [reply src/aerie_cli/aerie_host.py:275]: Understood! I was guessing it either did that or raised a generic Exception, thanks for confirming!

File: src/aerie_cli/aerie_host.py (hunk 2 above)
Comment (cartermak) [reply src/aerie_cli/aerie_host.py:29]: Thanks - I'm honestly not sure why I removed this. I wrote most of this months ago, so give it another look...

File: tests/unit_tests/test_aerie_host.py
Comment (cartermak) [reply tests/unit_tests/test_aerie_host.py:1]: No, this was a note to myself that I meant to remove. I'm taking a "slightly messy tests are better than no tests" position on this one for now.

File: src/aerie_cli/aerie_host.py (hunk 2 above)
Comment (Mythicaeda) [reply src/aerie_cli/aerie_host.py:29]: The `version` endpoint doesn't return `success`

File: tests/unit_tests/test_aerie_host.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    mock_response = MockResponse(json, text, ok)
+    mock_session = MockSession(mock_response)
+    return AerieHost("", "", mock_session)
+
+
+def test_check_aerie_version():
+    aerie_host = get_mock_aerie_host(
+        json={"version": COMPATIBLE_AERIE_VERSIONS[0]})
+
+    aerie_host.check_aerie_version()
```
Comment (Mythicaeda) [inline tests/unit_tests/test_aerie_host.py:46]: Is there no Python equivalent of an `assertDoesNotThrow` for unit tests?

Comment (cartermak) [reply tests/unit_tests/test_aerie_host.py:46]: Nope - tests that raise an error fail by default (at least with standard lib unittest and pytest), so no need to assert something doesn't raise an error.

File: src/aerie_cli/aerie_host.py (hunk 2 above)
Comment (cartermak) [reply src/aerie_cli/aerie_host.py]: Maybe why I did it originally, but it still only checked if the return value included the "success" key, so this branch isn't taken for a `get` request's response. Putting it back in and I'll see if anything breaks in the integration tests...

File: src/aerie_cli/utils/sessions.py (hunk 4 above)
Comment (cartermak) [reply src/aerie_cli/utils/sessions.py]: Agree, changed to `--force`

File: src/aerie_cli/aerie_host.py (hunk 2 above)
Comment (Mythicaeda) [reply src/aerie_cli/aerie_host.py]: Oh huh, you're right. Yeah, if it doesn't break anything, I'd say revert this change.

**Pull request merged**